runtime/bcutil/cfreader.c

```diff
@@ -528,7 +530,9 @@ readAttributes(J9CfrClassFile * classfile, J9CfrAttribute *** pAttributes, U_32
 
 				result = readAnnotations(classfile, parameterAnnotations->annotations,	parameterAnnotations->numberOfAnnotations, data, dataEnd,
 						segment, segmentEnd, &index, &freePointer, flags);
-				if (result != 0) {
+				if (BCT_ERR_OUT_OF_ROM == result) {
+					return result;
+				} else if (result != 0) {
```

Comment:
Can you change this to `(BCT_ERR_NO_ERROR != result)`?  Using the named error code is clearer and allows changing the value in the future

Reply:
Is this missing a `index != end` check?

Reply:
This check is inside a loop, so `index != end` isn't required as it will break out and be checked in more detail, see line 543 below.
I will update the 0 to `BCT_ERR_NO_ERROR`, thanks for pointing out

Reply:
Agreed.